packages/components/src/components/teaser/teaser.ts

```diff
@@ -0,0 +1,163 @@
+ * @slot meta - An optional meta slot.
+ * @slot headline - headline slot.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart media - The container that wraps the media.
+ * @csspart meta -  The container that wraps the meta.
+ * @csspart headline - The container that wraps the headline.
+ * @csspart main - The container that wraps the main content.
+ *
+ * @cssproperty --distribution-media - The distribution ratio of the media.
+ * @cssproperty --distribution-content - The distribution ratio of the content.
+ */
+
+@customElement('sd-teaser')
+export default class SdTeaser extends SolidElement {
+  @property({ reflect: true })
+  variant: 'white' | 'white border-neutral-300' | 'neutral-100' | 'primary' | 'primary-100' = 'white';
+
+  /** The teaser's orientation */
+  @property({ reflect: true })
+  orientation: 'responsive' | 'vertical' | 'horizontal' = 'responsive';
+
+  /** The teaser's padding */
+  @property({ type: Boolean, reflect: true }) inset = false;
+
+  @query('[part="base"]') teaser: HTMLElement;
+
+  @state() private currentOrientation: 'vertical' | 'horizontal';
+
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'media', 'meta', 'headline');
+
+  private resizeObserver: ResizeObserver;
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (window.ResizeObserver) {
+      this.resizeObserver = new ResizeObserver(() => this.updateOrientation());
+      this.resizeObserver.observe(this);
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  updateOrientation() {
+    if (this.orientation === 'responsive') {
+      this.currentOrientation = this.offsetWidth > 448 ? 'horizontal' : 'vertical';
+      this.requestUpdate();
+    } else {
+      this.currentOrientation = this.orientation;
+    }
+  }
+
+  render() {
+    if (this.variant === 'white border-neutral-300') {
+      this.inset = true;
+    }
+
+    const slots = {
+      default: this.hasSlotController.test('[default]'),
+      media: this.hasSlotController.test('media'),
+      meta: this.hasSlotController.test('meta'),
+      headline: this.hasSlotController.test('headline')
+    };
+
+    return html`
+      <div
+        class=${cx(
+          'flex',
+          {
+            white: 'bg-white',
+            'white border-neutral-300': 'bg-white border border-neutral-300',
+            'neutral-100': 'bg-neutral-100',
+            primary: 'bg-primary text-white',
+            'primary-100': 'bg-primary-100'
+          }[this.variant],
+          (this.orientation === 'vertical' || this.currentOrientation === 'vertical') && 'flex-col',
+          (this.orientation === 'horizontal' || this.currentOrientation === 'horizontal') && 'flex-row gap-8',
+          (this.orientation === 'horizontal' || this.currentOrientation === 'horizontal') && this.inset && 'py-8 px-10'
```

Comment:
You could always use `currentOrientation` if you set it once when initiating the component. It shoudl work in the constructor phase.